### PR TITLE
[ALLUXIO-1875] Add keyCount to PartitionInfo in key-value store

### PR DESCRIPTION
**Pull request by saltylin**
https://alluxio.atlassian.net/browse/ALLUXIO-1875

### CODE REVIEW

File: keyvalue/client/src/main/java/alluxio/client/keyvalue/KeyValuePartitionWriter.java [hunk 1]
```diff
@@ -66,4 +66,10 @@ public static KeyValuePartitionWriter create(AlluxioURI uri)
    * @return whether this writer is full to take any more key-value pairs
    */
   boolean canPut(byte[] key, byte[] value);
+
+  /**
+   * Returns the number of key-value pairs written to the partition.
+   * @return count of keys
```
Comment (uronce-cc) [inline keyvalue/client/src/main/java/alluxio/client/keyvalue/KeyValuePartitionWriter.java]: NIT: add empty line before this line

Comment (uronce-cc) [inline keyvalue/client/src/main/java/alluxio/client/keyvalue/KeyValuePartitionWriter.java]: Only need "@return <description>"

File: keyvalue/client/src/main/java/alluxio/client/keyvalue/BaseKeyValueStoreWriter.java [hunk 2]
```diff
@@ -181,10 +181,11 @@ private void completePartition() throws IOException, AlluxioException {
     if (mWriter == null) {
       return;
     }
+    int keyCount = mWriter.keyCount();
     mWriter.close();
     List<Long> blockIds = mFileSystem.getStatus(getPartitionName()).getBlockIds();
     long blockId = blockIds.get(0);
-    PartitionInfo info = new PartitionInfo(mKeyStart, mKeyLimit, blockId);
+    PartitionInfo info = new PartitionInfo(mKeyStart, mKeyLimit, blockId, keyCount);
```
Comment (uronce-cc) [inline keyvalue/client/src/main/java/alluxio/client/keyvalue/BaseKeyValueStoreWriter.java]: `mWriter.keyCount()`

**Comment by uronce-cc** [comment]
@saltylin Thanks for fixing the TODO! Left some comments.

File: keyvalue/client/src/main/java/alluxio/client/keyvalue/BaseKeyValueStoreWriter.java (hunk 2 above)
Comment (saltylin) [inline keyvalue/client/src/main/java/alluxio/client/keyvalue/BaseKeyValueStoreWriter.java]: Should we call `mWriter.keyCount()` even if it has been closed?

File: keyvalue/client/src/main/java/alluxio/client/keyvalue/KeyValuePartitionWriter.java (hunk 1 above)
Comment (saltylin) [inline keyvalue/client/src/main/java/alluxio/client/keyvalue/KeyValuePartitionWriter.java]: Updated!

**Comment by saltylin** [comment]
@uronce-cc Thanks for your review! I left a question in comments.

File: keyvalue/client/src/main/java/alluxio/client/keyvalue/BaseKeyValueStoreWriter.java (hunk 2 above)
Comment (uronce-cc) [inline keyvalue/client/src/main/java/alluxio/client/keyvalue/BaseKeyValueStoreWriter.java]: The information should still be available.

Comment (saltylin) [inline keyvalue/client/src/main/java/alluxio/client/keyvalue/BaseKeyValueStoreWriter.java]: @uronce-cc Updated!

**Comment by uronce-cc** [comment]
@saltylin Thanks for fixing the TODO! 

LGTM

**Comment by gpang** [comment]
@saltylin Overall this looks good. Thanks!

I just have one question. What is the reason for changing the count from a long to an int?

**Comment by saltylin** [comment]
@gpang Because `Index#keyCount()` returns an int. One KeyValuePartitionWriter is associated with one index, thus the types of the two keyCounts should be same.

**Comment by gpang** [comment]
@saltylin That makes sense.

Do you think it would be better to make everything a long then? I ask because you use an int32 for the thrift definition, and I'm wondering if it would be better to just make that thrift definition int64?

**Comment by saltylin** [comment]
@gpang Well, I think this is unnecessary. Let's assume the number of keys reaches the max value of int, which is 1 << 31 - 1. Every key-value pair is associated with a keyLength (int) and a valueLength (int), and just the length information will be about 1 << 34, that's 16GB, which should be far smaller than the real size since keys and values also consume bytes. Thus I think int is ok for the keyCount in a partition.

**Comment by gpang** [comment]
@apc999 can you take a look at the key count?

This PR looks good after that. Could you please resolve the merge conflicts? Thanks!

**Comment by saltylin** [comment]
@gpang Updated!

**Comment by gpang** [comment]
@apc999 could you take a look?

**Comment by gpang** [comment]
LGTM

**Comment by apc999** [comment]
LGTM

**Pull request merged**